src/group_theory/subgroup.lean

```diff
@@ -823,6 +823,12 @@ lemma gsmul_mem (H : add_subgroup A) {x : A} (hx : x ∈ H) :
 lemma sub_mem (H : add_subgroup A) {x y : A} (hx : x ∈ H) (hy : y ∈ H) : x - y ∈ H :=
 H.add_mem hx (H.neg_mem hy)
 
+lemma add_mem_iff_left (H : add_subgroup A) {x y : A} (hy : y ∈ H) : x + y ∈ H ↔ x ∈ H :=
+⟨λ hx, by simpa using sub_mem _ hx hy, λ hx, add_mem _ hx hy⟩
+
+lemma add_mem_iff_right (H : add_subgroup A) {x y : A} (hx : x ∈ H) : x + y ∈ H ↔ y ∈ H :=
+⟨λ hy, by simpa using H.add_mem (H.neg_mem hx) hy, add_mem _ hx⟩
```

Comment:
This proof changed slightly since it can no longer assume commutativity

Reply:
We already have these two lemmas. They're called `add_mem_cancel_left/right` (generated by `to_additive` from `mul_mem_cancel_left/right`).

Reply:
Good idea, sorry for the slow update. Annoying the left/right are swapped between those two theorems, but as you say, the conservative thing to do in this PR is not worry about it.

Reply:
Is this something the linter should have caught?